pkg/ast/spl/spl.peg

```diff
@@ -40,6 +40,130 @@ func ConvertCustomDateTimeFormatToEpochMs(dateStr string) (int64, error) {
     return t.UnixMilli(), nil
 }
 
+
+func applySnap(snap string, t time.Time) (time.Time, error) {
```

Comment:
It's pretty confusing reading this function unless you already know what `snap` should look like. Can you add a comment above this function to describe or give examples of what `snap` should contain?

Reply:
addressed in the latest commit